### ISSUE TITLE
Training using “mp.spawn”, can not reproduce the training results

### ISSUE THREAD

**Issue by konioyxgq** [pytorch/pytorch#39648]
I used a distributed training method. And use **mp.spawn(main_worker, nprocs=ngpus_per_node)**  to open up multiple processes.But the random number seeds for each process are different. This has led me to fail to reproduce previous results. My code is as follows: 

```
import torch
import torch.nn as nn
import torch.optim as optim
import torchvision.transforms as transforms
import torchvision.datasets as datasets
import numpy as np
import torch.multiprocessing as mp
import torch.distributed as dist

def main():
    SEED = 1375 # random seed for reproduce results
    torch.manual_seed(SEED)
    torch.backends.cudnn.deterministic = True
    torch.backends.cudnn.benchmark = True
    ngpus_per_node = torch.cuda.device_count()
    world_size = 1 * ngpus_per_node
    mp.spawn(main_worker, nprocs=ngpus_per_node)

def main_worker(gpu):
    ...
```

And the seeds I set in the main function don't seem to work.
What should I do to reproduce the previous results. Thank you！

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @xush6528 @osalpekar

**Comment by konioyxgq** [comment]
`import torch
import torch.nn as nn
import torch.optim as optim
import torchvision.transforms as transforms
import torchvision.datasets as datasets
import numpy as np
import torch.multiprocessing as mp
import torch.distributed as dist

def main():
    SEED = 1375 # random seed for reproduce results
    torch.manual_seed(SEED)
    torch.backends.cudnn.deterministic = True
    torch.backends.cudnn.benchmark = True
    ngpus_per_node = torch.cuda.device_count()
    world_size = 1 * ngpus_per_node
    mp.spawn(main_worker, nprocs=ngpus_per_node)

def main_worker(gpu):
    ...`

**Comment by SsnL** [comment]
seed in worker function, not main. the process that runs training never executed main.

**Comment by konioyxgq** [comment]
Okay, I got it. Thank you for your reply.

**Comment by mrshenli** [comment]
Closing this issue for now. Please feel free to reopen if anything needs to be fixed.
